rootfs/etc/nginx/template/nginx.tmpl

```diff
@@ -68,70 +68,22 @@ http {
 
     {{ buildLuaSharedDictionaries $cfg $servers }}
 
-    init_by_lua_block {
-        collectgarbage("collect")
-
-        -- init modules
-        local ok, res
-
-        ok, res = pcall(require, "lua_ingress")
-        if not ok then
-          error("require failed: " .. tostring(res))
-        else
-          lua_ingress = res
-          lua_ingress.set_config({{ configForLua $all }})
-        end
-
-        ok, res = pcall(require, "configuration")
-        if not ok then
-          error("require failed: " .. tostring(res))
-        else
-          configuration = res
-          configuration.prohibited_localhost_port = '{{ .StatusPort }}'
-        end
-
-        ok, res = pcall(require, "balancer")
-        if not ok then
-          error("require failed: " .. tostring(res))
-        else
-          balancer = res
-        end
-
-        {{ if $all.EnableMetrics }}
-        ok, res = pcall(require, "monitor")
-        if not ok then
-          error("require failed: " .. tostring(res))
-        else
-          monitor = res
-        end
-        {{ end }}
+    lua_shared_dict luaconfig 5m;
 
-        ok, res = pcall(require, "certificate")
-        if not ok then
-          error("require failed: " .. tostring(res))
-        else
-          certificate = res
-          certificate.is_ocsp_stapling_enabled = {{ $cfg.EnableOCSP }}
-        end
-
-        ok, res = pcall(require, "plugins")
-        if not ok then
-          error("require failed: " .. tostring(res))
-        else
-          plugins = res
-        end
-        -- load all plugins that'll be used here
-        plugins.init({ {{ range  $idx, $plugin := $cfg.Plugins }}{{ if $idx }},{{ end }}{{ $plugin | quote }}{{ end }} })
+    {{/* We need to keep this lua block inline, because init_worker_by_lua_file does not support using arguments */}}
+    init_by_lua_block {
+        local luaconfig = ngx.shared.luaconfig
+        local ingresscfg =  {{ configForLua $all }}
+        luaconfig:set("enablemetrics", {{ $all.EnableMetrics }})
+        luaconfig:set("listen_https_ports", '{{ $all.ListenPorts.HTTPS }}')
+        luaconfig:set("use_forwarded_headers", {{ $cfg.UseForwardedHeaders }})
+        local ngx_conf_init = require('ngx_conf_init')
+        ngx_conf_init.initialize_ingress('{{ .StatusPort }}', {{ $all.EnableMetrics }}, {{ $cfg.EnableOCSP }}, ingresscfg)
     }
 
     init_worker_by_lua_block {
-        lua_ingress.init_worker()
-        balancer.init_worker()
-        {{ if $all.EnableMetrics }}
-        monitor.init_worker({{ $all.MonitorMaxBatchSize }})
-        {{ end }}
-
-        plugins.run()
+        local ngx_conf_init_worker = require('ngx_conf_init_worker')
+        ngx_conf_init_worker.initialize_worker({{ $all.EnableMetrics }}, {{ $all.MonitorMaxBatchSize }})
     }
```

Comment:
This may not be something desirable, but for these global init blocks if we templated lua files we could them load them by `init_by_lua_file` and `init_worker_by_lua_file` and be able to get rid of the inlining here.

Reply:
Can work. Another thing we can make is to write the desired configs on some json and make the script read for it. My concern is that I am not sure how much we rely on changes on template or lua files to trigger a reload, this can be a problem (still, not sure)